src/hooks/use-list-behavior.js

```diff
@@ -0,0 +1,72 @@
+import { useState, useCallback } from 'react'
+
+/**
+ * Hook that returns bind props + accessibility props for
+ * "selectable" elements
+ */
+
+const resetStyles = {
+  outline: 'none'
+}
+
+export function useListBehavior({
+  disabled,
+  isHighlighted,
+  isSelectable,
+  isSelected,
+  onSelect,
+  ref: inputRef
+}) {
+  const [ref, setRef] = useState(inputRef)
+
+  const keyDownHandler = useCallback(
+    e => {
+      const { key } = e
+      if (ref) {
+        let nextItemToFocus
+        const tableBodyChildren = Array.from(ref.parentElement.children)
+        const rowIndex = tableBodyChildren.indexOf(ref)
+
+        if (key === 'ArrowUp' && rowIndex - 1 >= 0) {
+          nextItemToFocus = tableBodyChildren[rowIndex - 1]
+        } else if (
+          key === 'ArrowDown' &&
+          rowIndex + 1 < tableBodyChildren.length
+        ) {
+          nextItemToFocus = tableBodyChildren[rowIndex + 1]
+        }
```

Comment:
If we are implementing arrow key navigation we should be using the roving tabindex pattern here – which this is not.

TLDR: only the next focused item should have a `tabIndex` value set, and all other items should have `tabIndex={-1}`.